cmd/neofs-cli/modules/object/delete.go

```diff
@@ -28,53 +28,67 @@ func initObjectDeleteCmd() {
 	flags := objectDelCmd.Flags()
 
 	flags.String(commonflags.CIDFlag, "", commonflags.CIDFlagUsage)
-	flags.String(commonflags.OIDFlag, "", commonflags.OIDFlagUsage)
+	flags.StringSlice(commonflags.OIDFlag, nil, commonflags.OIDFlagUsage)
 	flags.Bool(binaryFlag, false, "Deserialize object structure from given file.")
 	flags.String(fileFlag, "", "File with object payload")
+
+	_ = objectDelCmd.MarkFlagRequired(commonflags.CIDFlag)
+	_ = objectDelCmd.MarkFlagRequired(commonflags.OIDFlag)
 }
 
 func deleteObject(cmd *cobra.Command, _ []string) {
 	ctx, cancel := commonflags.GetCommandContext(cmd)
 	defer cancel()
 
 	var cnr cid.ID
-	var obj oid.ID
-	var objAddr oid.Address
+	var objs []oid.ID
+	var objAddrs []oid.Address
 
 	binary, _ := cmd.Flags().GetBool(binaryFlag)
 	if binary {
 		filename, _ := cmd.Flags().GetString(fileFlag)
 		if filename == "" {
 			common.ExitOnErr(cmd, "", fmt.Errorf("required flag \"%s\" not set", fileFlag))
 		}
-		objAddr = readObjectAddressBin(cmd, &cnr, &obj, filename)
+		var obj oid.ID
+		objAddr := readObjectAddressBin(cmd, &cnr, &obj, filename)
+
+		objs = []oid.ID{obj}
+		objAddrs = []oid.Address{objAddr}
 	} else {
-		cidVal, _ := cmd.Flags().GetString(commonflags.CIDFlag)
-		if cidVal == "" {
-			common.ExitOnErr(cmd, "", fmt.Errorf("required flag \"%s\" not set", commonflags.CIDFlag))
-		}
+		readCID(cmd, &cnr)
+		oIDVals, _ := cmd.Flags().GetStringSlice(commonflags.OIDFlag)
```

Comment:
lets check `len(oIDVals) > 0`, seems like `objectDelCmd.MarkFlagRequired(commonflags.OIDFlag)` is not enough
```
$ neofs-cli object delete --cid F74twfixT89Wqc4qh32Yj5vzP6p9gQ7dxJCzp3C311an 
Error: required flag(s) "oid" not set
$ neofs-cli object delete --cid F74twfixT89Wqc4qh32Yj5vzP6p9gQ7dxJCzp3C311an --oid '' --oid '' && echo $?
0
```

#2879